File: js/fizzbuzz_map.js
fizzBuzzMap = (length) => {
    // create an array filled with numbers up to length
    const numbers = Array.apply(null, {length}).map(Number.call, Number)

    // run check for all numbers and return an array with the results
    return numbers.map(divisibleCheck)
}

// check if a number is cleanly divisible by 5, 3 or 15(5 and 3)
const divisibleCheck = (num) => {
    if(num % 15 === 0) {
        return 'FizzBuzz'
    } else if (num % 3 === 0) {
        return 'Fizz'
    } else if (num % 5 === 0) {
        return 'Buzz'
    } else {
        return num
    }
}

fizzBuzzMap(100)

File: java/FizzBuzzMap.java
import java.util.HashMap;
import java.util.TreeMap;

public class FizzBuzzMap{

	public static void main(String []args) {
		HashMap<Integer,String> fizzBuzzTreeMap = new HashMap<Integer, String>();
		for (int i = 1; i <= 100; i++) {
			if (i % 5 ==0 && i % 3 == 0) {
				fizzBuzzTreeMap.put(i, "FizzBuzz");
			} else if (i % 5 == 0) {
				fizzBuzzTreeMap.put(i, "Buzz");
			} else if (i % 3 ==0) {
				fizzBuzzTreeMap.put(i, "Fizz");
			} else {
				fizzBuzzTreeMap.put(i, String.valueOf(i));
			}
		}
		
		for (int i = 1; i <= 100; i++) {
			System.out.println(fizzBuzzTreeMap.get(i));
		}
	}
}